saleor/checkout/migrations/0057_remove_metadata_from_checkout_model_state.py

```diff
@@ -0,0 +1,47 @@
+from django.db import migrations
+import django.contrib.postgres.indexes
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("checkout", "0056_move_checkout_metadata_to_separate_model"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            # Delete fields from state only, removal from db will happen in
+            # next patch to prevent downtime and ensure backward-compatibility
+            state_operations=[
+                migrations.RemoveField(
+                    model_name="checkout",
+                    name="metadata",
+                ),
+                migrations.RemoveField(
+                    model_name="checkout",
+                    name="private_metadata",
+                ),
+            ],
+        ),
+        migrations.RemoveIndex(
+            model_name="checkout",
+            name="checkout_p_meta_idx",
+        ),
+        migrations.RemoveIndex(
+            model_name="checkout",
+            name="checkout_meta_idx",
+        ),
+        migrations.AddIndex(
+            model_name="checkoutmetadata",
+            index=django.contrib.postgres.indexes.GinIndex(
+                fields=["private_metadata"], name="checkoutmetadata_p_meta_idx"
```

Comment:
What do we think this index would be for? Is it just for filtering the checkouts? If so maybe we could add a combined one instead of two separate ones (unless there is a case of filtering by ONLY meta OR only private meta which I'm unaware). If there is another case for more than filtering or separate filtering, ignore this comment.

Reply:
We should not mix the two. They are separate namespaces, they can share the same keys and values, and you should filter them separately.

Reply:
Adding to the question of @tomaszszymanski129: what is it used for? Isn't `private_metadata` JSONB? Is the indexing brought by `JSONB` not enough?

Reply:
I created indexes to stay consistent since they are present in `ModelWithMetadata` for `metadata` and `private_metadata` fields. I think that even though Checkout metadata is in separate model the functionality should stay the same between  `ModelWithMetadata` and `CheckoutMetadata`